### ISSUE TITLE
ENH: outlier-influence leave-on-out statistics for variable selection

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#7602]
This is just an idea

We don't directly support generic cross-validation, i.e. pipepline as in sklearn.
We have approximate leave-one-out statistics and related statistics, aic, gcv, computed without cross-val loop.
These statistics like hat_value, df_betas are computed mostly in outlier_influence analysis.
Currently, the main/only focus of outlier-influence code are outlier and influence measures.

Can we use statistics from outlier_influence analysis for variable selection and hyperparameter selection?
e.g. is there a usage for dfbetas?
(relate to that? : #6323 #6346 sensitivity and robustness of estimates for parameter of interest. )

E.g. 
- loop over model candidates computing statistics to compare and evaluate models
- compute relevant outlier influence statistics
- provide summary measures for the models based on approximate LOOO statistics

We also have an (unused?) leave-one-variable-out statistic in some outlier-influence classes.

We do something similar already when we compute gcv for hyper parameter selection in penalized models like GAM.
In those cases we compute hat_matrix_diagonal/influence already in the model and don't need to go through outlier-influence analysis.

Currently, outlier-influence statistics are computed for the consumption by users. 
We should take advantage of it also in internal use, to provide more than those statistics to users.
